src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -2,13 +2,160 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Diagnostics;
-
+using System.Numerics;
+using System.Runtime.CompilerServices;
 using Internal.Runtime.CompilerServices;
 
 namespace System
 {
     internal static partial class SpanHelpers // .T
     {
+        public static void Fill<T>(ref T refData, uint numElements, T value)
+        {
+            // n.b. If Fill is ever adapted to work for lengths beyond uint.MaxValue, double-check
+            // where it's used in the arithmetic operations below to ensure no integer overflow is possible.
+
+            if (numElements == 0)
+            {
+                return; // nothing to do
+            }
+
+            if (typeof(T) != typeof(float) && typeof(T) != typeof(double) && !RuntimeHelpers.IsBitwiseEquatable<T>())
```

Comment:
So half is not optimized? Maybe add it to your Benchmarks anyway?

Reply:
I'd definitely appreciate us adding `Half` support here. We'll be adding supporting to the vector types and elsewhere eventually.

It might even be nice if we had some assert we could put in places like this to help ensure "primitive" types are handled. Off the top of my head these are `byte`, `sbyte`, short`, `ushort`, `int`, `uint`, `long`, `ulong`, `float`, and `double`.
There is also, as of recent C# versions, `nint` and `nuint`; and `Half` as of the latest .NET versions.

If you look a bit beyond that for other "primitives" or language special types (not just those that are directly primitive and have hardware acceleration for SIMD scenarios), there is also `char`, `bool`, `decimal`, and maybe a few others.